### PR TITLE
[Enhancement] Enable fast schema evolution for rollup

### PR DESCRIPTION
**Pull request by sevev**
Why I'm doing:

If we create a table enable fast schema evolution, add a new column task will run as a fast schema evolution task but now the original linked schema change task. However, if we add a new column to rollup, the task still run as linked schema change task. But these operations could be run as fast schema evolution task.

What I'm doing:

Enable fast schema evolution for rollup.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -2397,16 +2396,13 @@ public void modifyTableAddOrDropColumns(Database db, OlapTable olapTable,
                     }
                 }
                 currentIndexMeta.setSchema(indexSchema);
-                if (!newSortKeyIdxes.isEmpty()) {
-                    currentIndexMeta.setSortKeyIdxes(newSortKeyIdxes);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:2401]: why remove?

File: be/src/storage/delta_writer.cpp [hunk 2]
```diff
@@ -544,11 +544,12 @@ Status DeltaWriter::_build_current_tablet_schema(int64_t index_id, const POlapTa
...
-        if (ptable_schema_param.indexes_size() > 0 && ptable_schema_param.indexes(i).has_column_param() &&
-            ptable_schema_param.indexes(i).column_param().columns_desc_size() != 0 &&
-            ptable_schema_param.indexes(i).column_param().columns_desc(0).unique_id() >= 0) {
+        auto& index = ptable_schema_param.indexes(i);
+        if (index.has_column_param() && index.column_param().columns_desc_size() != 0 &&
+            index.column_param().columns_desc(0).unique_id() >= 0 && index.has_schema_version() &&
+            index.schema_version() > _tablet_schema->schema_version()) {
             RETURN_IF_ERROR(_tablet_schema->build_current_tablet_schema(
-                    index_id, ptable_schema_param.version(), ptable_schema_param.indexes(i), ori_tablet_schema));
+                    index_id, ptable_schema_param.indexes(i).schema_version(), index, ori_tablet_schema));
```
[the remark this answers is not in the document]
Comment (sevev) [reply be/src/storage/delta_writer.cpp]: done, changed

File: be/src/exec/tablet_info.h [hunk 3]
```diff
@@ -47,6 +47,8 @@ struct OlapTableIndexSchema {
     int32_t schema_hash;
     OlapTableColumnParam* column_param;
     ExprContext* where_clause = nullptr;
+    int32_t schema_version = -1;
+    ;
```
[the remark this answers is not in the document]
Comment (sevev) [reply be/src/exec/tablet_info.h]: changed.

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java (hunk 1 above)
Comment (sevev) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:2401]: > why remove?

These should not be removed, changed.

**Review by wyb — DISMISSED** [review]

**Review by decster — COMMENTED** [review]
does it apply to all data models?  cause aggregate and unique key cannot just fill a default value to newly added column in roll up table.

**Pull request closed without merging**